lib/Doctrine/DBAL/Platforms/SqlitePlatform.php

```diff
@@ -345,7 +346,10 @@ protected function _getCreateTableSQL($name, array $columns, array $options = []
 
         if (isset($options['primary']) && ! empty($options['primary'])) {
             $keyColumns = array_unique(array_values($options['primary']));
-            $queryFields.= ', PRIMARY KEY('.implode(', ', $keyColumns).')';
+            // the primary key is already defined in the column definition for auto increment fields
+            if (count($keyColumns) > 1 || ! $columns[$keyColumns[0]]['autoincrement']) {
+                $queryFields .= ', PRIMARY KEY(' . implode(', ', $keyColumns) . ')';
```

Comment:
Do we still need to use two syntaxes? The `PRIMARY KEY(x, y, z)` looks more generic since it can describe multiple fields. Given that we take this case into account, should we use just the generic syntax.

This `$columns[$keyColumns[0]]` doesn't look good. What if the key consists of two columns where the second one has autoincrement?

Reply:
If that's the case, can we leave the `AUTOINCREMENT` keyword with the field and define the PK later? Otherwise, it's unclear if it's possible and how it will be handled if one of the columns is autoincremented but the PK consists of two columns.

Reply:
Unfortunately, it only works when the field definition is `INTEGER PRIMARY KEY AUTOINCREMENT` as a whole. In that specific order.

Reply:
Still, we need to figure out how this is going to be handled if at all possible:
> one of the columns is autoincremented but the PK consists of two columns.

The `0` in `count($keyColumns) > 1 || ! $columns[$keyColumns[0]]['autoincrement']` looks like a magic constant.

Reply:
I agree. I'll run some tests to see if I can improve that.

Reply:
I've made some improvements. This should cover it.